### DISCUSSION TITLE
Git-GitHub flow

### DISCUSSION THREAD

**Discussion by fromITzerotoIThero** [fromITzerotoIThero/personal-website#11] [General]
Oh W.! 
I wanted to create the redirect rules for Netlify, and so I did, on a new branch `feature/netlify-redirects`, made a PR, but since it passed all of the `deployment` tests and (for some unknown reason...) there was no review required, I was like "it's just one tiny little file, naaa, I'm not gonna bother Wici about this!", I've merged it with master, than deleted the remote branch and...

here's where the trouble(?) starts:

- I wanted to keep branches clean and so I also deleted the `feature/netlify-redirects` branch locally in VS console
- and that's what I got:

![redirects](https://github.com/fromITzerotoIThero/personal-website/assets/109156809/27765197-c898-4a01-b03b-45caa7e7d1d0)

If my local master is up-to-date with remote master, then why is the name of the file crossed-out?
Can I safely work making new branches from the version of master that I have locally?
(The website works fine after all of this).

**Comment by piotrwicijowski** [comment]
I have a strong suspicion that your local master is not up to date with the actual remote master (only its assumption about origin/master). Did you make sure to pull?

**Comment by fromITzerotoIThero** [comment]
I've just pulled for my local master. and it says it's up to date. I can't easily check if works, because it seems that the file explorers hide such files... 
Anyway, I've made a new branch `style/cosmetic-improvements`, I'm working on it yet probably I'll make a PR soon have enough time to handle fuckups XP

**Comment by fromITzerotoIThero** [comment]
Dear W.,

So I wanted to change favicons on personal website. 
- I switched to local master in order to pull latest changes (there were a lot - the hearts and fixing of the scrollbars on `ProjectPage`)
- I tried to fire the development server, but there's an error when on master branch 
- I don't quite understand why it throws these errors - imports in the given files seem fine to me 🤷‍♀️  
- I didn't want to create a new branch from this malfunctioning local master so as not to screw things further

![master-personal-website](https://github.com/fromITzerotoIThero/personal-website/assets/109156809/9e0039fa-d204-406d-a82c-2739e422b011)

My questions are:
1. How come there's an error on my local master (up-to-date with origin/master) when the website (reflecting the remote master branch) is running perfectly fine?
2. Is it safe to leave the local master as is and checkout for a new branch from it or is it wiser to fix the local master first, prior to making any new branches?

**Reply by piotrwicijowski** [reply]
Did you do `npm install`?

**Comment by fromITzerotoIThero** [comment]
Oh W., 

If you have a minute, would you please take a look at this [closed PR ](https://github.com/fromITzerotoIThero/personal-website/pull/17)?

The new favicon doesn't show:
- on my local `enhancement/favicon-change` branch, graphic files represent the new version, yet while running the site on the development server they don't show
- when I made that PR, netlify preview showed me the right one
- yet when I merged the branch with `master` - it's still shows the old version.

I have a feeling it might have something to do with the way I switched those files (manually in the file explorer) and that they had exactly the same name.

My question is:
1. should I pull for my local master to stay up-to-date with remote and simply make a new branch, switch files (but this time make them distinguishable with different names) and make a new PR? 
2. or I'm I missing something here that can get me in some Git-GitHub mess again? 🤠

**Reply by piotrwicijowski** [reply]
Hmm, weird, works for me. I'm guessing that it may be that your browser has cached the favicon. Maybe just opening it in a different browser or in incognito mode should allow you to see the change, even on the local branch

**Reply by fromITzerotoIThero** [reply]
Ha! It was caching indeed! Thank you!!!
